src/app/tracing/graph/components/graph-legend-view/graph-legend-view.component.ts

```diff
@@ -46,47 +39,13 @@ export class GraphLegendViewComponent {
         return this.legend.length === 0;
     }
 
-    private updateLegend(legendInfo: LegendInfo) {
-        const newLegend: LegendEntryWithIndices[] = [];
-        if (legendInfo) {
-            legendInfo.stations.forEach((stationLegendInfo, index) => {
-                newLegend.push({
-                    name: stationLegendInfo.label,
-                    stationColor: stationLegendInfo.color ?? undefined,
-                    shape: stationLegendInfo.shape ?? undefined
-                });
-            });
-            const entryMap = Utils.createObjectFromArray(newLegend, (e) => e.name);
-            legendInfo.deliveries.forEach((deliveryLegendInfo, index) => {
-                const entry = entryMap[deliveryLegendInfo.label];
-                if (entry) {
-                    entry.deliveryColor = deliveryLegendInfo.color ?? undefined;
-                    entry.deliveryIndex = index;
-                } else {
-                    const newEntry: LegendEntryWithIndices = {
-                        name: deliveryLegendInfo.label,
-                        deliveryColor: deliveryLegendInfo.color ?? undefined,
-                        deliveryIndex: index
-                    };
-                    const beforeIndex = newLegend.findIndex(e => e.deliveryIndex !== undefined && e.deliveryIndex > index);
-
-                    if (beforeIndex >= 0) {
-                        newLegend.splice(beforeIndex, 0, newEntry);
-                    } else {
-                        newLegend.push(newEntry);
-                    }
-                }
-            });
-        }
 
-        this.legend = newLegend.map(e => ({
-            name: e.name,
-            stationColor: e.stationColor,
-            deliveryColor: e.deliveryColor,
-            shape: e.shape
-        }));
+    private updateLegend(legend: LegendDisplayEntry[]) {
```

Comment:
This function is somehow inconsequent. Line 44 considers the case that `legend` is `undefined` or `null`.
Problem 1: argument type does not allow such cases
Problem 2: the legend is not updated if legend is undefined or null